mlir/lib/Dialect/AMDGPU/Transforms/OptimizeSharedMemory.cpp

```diff
@@ -218,7 +218,8 @@ mlir::amdgpu::optimizeSharedMemoryReadsAndWrites(Operation *parentOp,
   return success();
 }
 
-void amdgpu::optimizeSharedMemoryReadsAndWritesOp(func::FuncOp funcOp) {
+std::optional<mlir::LogicalResult>
```

Comment:
I don't see why the result is optional, nothing returns `nullopt` here.

Reply:
There is one more `return;` statement above after checking `hasSharedMemoryAddressSpace()`. There is no failure in this case it just checks if the shared memory relation exists or not. So I added an optional for that.